airflow/models/baseoperator.py

```diff
@@ -433,8 +431,7 @@ def __init__(
             self._outlets.update(outlets)
 
     def __eq__(self, other):
-        if (type(self) == type(other) and  # pylint: disable=unidiomatic-typecheck
-                self.task_id == other.task_id):
+        if type(self) is type(other) and self.task_id == other.task_id:
```

Comment:
I think this changes the behavior:

>The **_==_** operator compares the values of both the operands and checks for value equality. Whereas **_is_** operator checks whether both the operands refer to the same object or not

Reply:
In case of type() objects it's a bit faster and works the same way. Types are always singletons   so == is equivalent to "is". It's - I think - common practice and it is an explicit way of comparing "exact" types rather than using isinstance. At least pylint does not complain any more with "suspicious type comparision".

Reply:
Yes you are right. Makes sense